packages/babylon/test/fixtures/typescript/cast/assert-and-assign/expected.json

```diff
@@ -0,0 +1,435 @@
+                  },
+                  "identifierName": "age"
+                },
+                "name": "age"
+              },
+              "computed": false,
+              "typeAnnotation": {
+                "type": "TSTypeAnnotation",
+                "start": 45,
+                "end": 53,
+                "loc": {
+                  "start": {
+                    "line": 3,
+                    "column": 5
+                  },
+                  "end": {
+                    "line": 3,
+                    "column": 13
+                  }
+                },
+                "typeAnnotation": {
+                  "type": "TSNumberKeyword",
+                  "start": 47,
+                  "end": 53,
+                  "loc": {
+                    "start": {
+                      "line": 3,
+                      "column": 7
+                    },
+                    "end": {
+                      "line": 3,
+                      "column": 13
+                    }
+                  }
+                }
+              }
+            }
+          ]
+        }
+      },
+      {
+        "type": "VariableDeclaration",
+        "start": 57,
+        "end": 70,
+        "loc": {
+          "start": {
+            "line": 6,
+            "column": 0
+          },
+          "end": {
+            "line": 6,
+            "column": 13
+          }
+        },
+        "declarations": [
+          {
+            "type": "VariableDeclarator",
+            "start": 61,
+            "end": 69,
+            "loc": {
+              "start": {
+                "line": 6,
+                "column": 4
+              },
+              "end": {
+                "line": 6,
+                "column": 12
+              }
+            },
+            "id": {
+              "type": "Identifier",
+              "start": 61,
+              "end": 69,
+              "loc": {
+                "start": {
+                  "line": 6,
+                  "column": 4
+                },
+                "end": {
+                  "line": 6,
+                  "column": 12
+                },
+                "identifierName": "response"
+              },
+              "name": "response"
+            },
+            "init": null
+          }
+        ],
+        "kind": "let"
+      },
+      {
+        "type": "ExpressionStatement",
+        "start": 71,
+        "end": 131,
+        "loc": {
+          "start": {
+            "line": 7,
+            "column": 0
+          },
+          "end": {
+            "line": 10,
+            "column": 1
+          }
+        },
+        "expression": {
+          "type": "AssignmentExpression",
+          "start": 71,
+          "end": 131,
+          "loc": {
+            "start": {
+              "line": 7,
+              "column": 0
+            },
+            "end": {
+              "line": 10,
+              "column": 1
+            }
+          },
+          "operator": "=",
+          "left": {
```

Comment:
This seems incorrect, I would expect a `TSAsExpression` node here.

Reply:
I looked into it. I think I've misunderstood the problem. I was supposed to just support parsing `TSAsExpression` in the provided input code but I ended up transforming it (which is meant to be done [here](https://github.com/babel/babel/blob/master/packages/babel-plugin-transform-typescript/src/index.js#L228) I suppose).

Can you point me to an example of creating a new node so that I could do that here as well?

Reply:
I think I figured it out. In `toAssignable`, I don't need to go deep into the nested `expression`. Just do the following:

```js
case "TSAsExpression":
  return node;
```
Does that make sense?

Reply:
I think you should convert the expression:
```
case TSAsExpression:
  node.expression = this.toAssignable(node.expression);
  return node;
```